// @ant-design-vue/pro-layout/es/WrapContent.js
import PropTypes from 'ant-design-vue/es/_util/vue-types';
import { ConfigProvider, Layout } from 'ant-design-vue';
import GridContent from './components/GridContent';

var Content = Layout.Content;


var WrapContentProps = {
  isChildrenLayout: PropTypes.bool,
  location: PropTypes.any,
  contentHeight: PropTypes.number,
  contentWidth: PropTypes.bool
};

var WrapContent = {
  name: 'WrapContent',
  props: WrapContentProps,
  render: function render(h) {
    var _$props = this.$props,
        isChildrenLayout = _$props.isChildrenLayout,
        contentWidth = _$props.contentWidth;

    return h(Content, [h(
      ConfigProvider,
      {
        attrs: {
          getPopupContainer: function getPopupContainer(el, dialogContext) {
            if (isChildrenLayout) {
              return el.parentNode();
            }
            return document.body;
          }
        }
      },
      [h(
        'div',
        { 'class': 'ant-pro-basicLayout-children-content-wrap' },
        [h(
          GridContent,
          {
            attrs: { contentWidth: contentWidth }
          },
          [this.$slots['default']]
        )]
      )]
    )]);
  }
};

export default WrapContent;
